pkg/controller/operators/catalog/operator.go

```diff
@@ -1654,11 +1693,59 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				plan.Status.Plan[i].Status = status
 
 			default:
-				return v1alpha1.ErrInvalidInstallPlan
-			}
+				if !isSupported(step.Resource.Kind) {
+					// Not a supported resource
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusUnsupportedResource
+					return v1alpha1.ErrInvalidInstallPlan
+				}
+
+				// Marshal the manifest into an unstructured object
+				dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(step.Resource.Manifest), 10)
+				unstructuredObject := &unstructured.Unstructured{}
+				if err := dec.Decode(unstructuredObject); err != nil {
+					return errorwrap.Wrapf(err, "error decoding %s object to an unstructured object", step.Resource.Name)
+				}
+
+				// Get the resource from the GVK.
+				gvk := unstructuredObject.GroupVersionKind()
+				r, err := o.apiresourceFromGVK(gvk)
+				if err != nil {
+					return err
+				}
+
+				// Create the GVR
+				gvr := schema.GroupVersionResource{
+					Group:    gvk.Group,
+					Version:  gvk.Version,
+					Resource: r.Name,
+				}
+
+				// Set up the dynamic client ResourceInterface
+				var resourceInterface dynamic.ResourceInterface
+				if r.Namespaced {
+					ownerutil.AddOwner(unstructuredObject, plan, false, false)
+					unstructuredObject.SetNamespace(namespace)
+					resourceInterface = o.dynamicClient.Resource(gvr).Namespace(namespace)
+				} else {
+					resourceInterface = o.dynamicClient.Resource(gvr)
+				}
+
+				// Update UIDs on all CSV OwnerReferences
+				updated, err := o.getUpdatedOwnerReferences(unstructuredObject.GetOwnerReferences(), plan.Namespace)
+				if err != nil {
+					return errorwrap.Wrapf(err, "error generating ownerrefs for unstructured object: %s", unstructuredObject.GetName())
+				}
+				unstructuredObject.SetOwnerReferences(updated)
 
+				// Ensure Unstructured Object
+				status, err := ensurer.EnsureUnstructuredObject(resourceInterface, unstructuredObject)
+				if err != nil {
+					return err
+				}
+
+				plan.Status.Plan[i].Status = status
+			}
 		default:
```

Comment:
Now that you no longer return an error here, this default statement is useless. Maybe you should remove it altogether?

On the other hand, I wonder what we should do if the Step doesn't have a status. Maybe we should change to `StepStatusUnknown` so it can be processed later. Would that be an appropriate behavior @ecordell?

Reply:
After reviewing the switch block - we should probably continue to return the error.

The only case we should hit this issue is when the installPlan includes an Unsupported Resource, where we'd wan't it to fail anyways.